llpc/util/llpcCacheAccessor.cpp

```diff
@@ -178,17 +178,17 @@ bool CacheAccessor::lookUpInShaderCache(const MetroHash::Hash &hash, bool alloca
 void CacheAccessor::setElfInCache(BinaryData elf) {
   if (m_shaderCacheEntryState == ShaderEntryState::Compiling && m_shaderCacheEntry) {
     updateShaderCache(elf);
-    m_shaderCache->retrieveShader(m_shaderCacheEntry, &m_elf.pCode, &m_elf.codeSize);
+    m_cacheResult = m_shaderCache->retrieveShader(m_shaderCacheEntry, &m_elf.pCode, &m_elf.codeSize);
     m_shaderCacheEntryState = ShaderEntryState::Ready;
   }
 
   if (!m_cacheEntry.IsEmpty()) {
+    m_cacheResult = Result::ErrorUnknown;
     if (elf.pCode) {
-      m_cacheEntry.SetValue(true, elf.pCode, elf.codeSize);
-      m_cacheEntry.GetValueZeroCopy(&m_elf.pCode, &m_elf.codeSize);
+      if (m_cacheEntry.SetValue(true, elf.pCode, elf.codeSize) == Result::Success)
+        m_cacheResult = m_cacheEntry.GetValueZeroCopy(&m_elf.pCode, &m_elf.codeSize);
```

Comment:
I'm not sure what should happen here. @s-perron?

Reply:
To avoid regression, I re-edit this piece of code by checking the result instead of changing its logic.